packages/core/src/InstanceSettings.ts

```diff
@@ -86,6 +86,29 @@ export class InstanceSettings {
 	 */
 	readonly hostId: string;
 
+	private isMultiMainEnabled = false;
+
+	private isMultiMainLicensed = false;
+
+	/** Set whether multi-main mode is enabled. Does not imply licensed status. */
+	setMultiMainEnabled(newState: boolean) {
+		this.isMultiMainEnabled = newState;
+	}
+
+	setMultiMainLicensed(newState: boolean) {
+		this.isMultiMainLicensed = newState;
+	}
+
+	/** Whether this `main` instance is running in multi-main mode. */
+	get isMultiMain() {
+		return this.instanceType === 'main' && this.isMultiMainEnabled && this.isMultiMainLicensed;
+	}
+
+	/** Whether this `main` instance is running in single-main mode. */
+	get isSingleMain() {
```

Comment:
I'm conflicted about if we need this additional getter, instead of just using `!instanceSettings.isMultiMain`. 
I get that this might make some code more readable, but I also find it a bit confusing when I look at distinct `isSingleMain` and `isMultiMain`, because I can't be sure on a first glance if these two flags are independent of each other or not.

Reply:
Never considered this. If you feel strongly about it let me know and I'll remove this one getter. Meanwhile I'll merge to unblock the rest of the work and prevent conflicts.